### PR TITLE
NAS-117398 / 22.12 / Storj iX cloud sync service

### PR DESCRIPTION
**Pull request by themylogin**

### CODE REVIEW

**Comment by themylogin** [comment]
time 3:00

File: src/middlewared/middlewared/rclone/remote/storjix.py [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+    rclone_type = "s3"
+
+    credentials_schema = [
+        Str("access_key_id", title="Access Key ID", required=True),
+        Str("secret_access_key", title="Secret Access Key", required=True),
+    ]
+
+    task_schema = []
+
+    async def list_buckets(self, credentials):
```
Comment (yocalebo) [inline src/middlewared/middlewared/rclone/remote/storjix.py:30]: Your outer and inner methods are named the same. Please change name of inner method to something else or move it into its own method.

**Review by yocalebo — CHANGES_REQUESTED** [review]

File: src/middlewared/middlewared/rclone/remote/storjix.py (hunk 1 above)
Comment (themylogin) [reply src/middlewared/middlewared/rclone/remote/storjix.py:30]: Will do that, but just curious why can't they be names the same?

Comment (yocalebo) [reply src/middlewared/middlewared/rclone/remote/storjix.py:30]: Just seems strange to name it the same. When I see methods that are named the same, I'm thinking of method overwriting when inheriting from a parent or super class.

**Comment by william-gr** [comment]
Was this tested using an account with the proper attribution? Just to make sure. Wasn't sure if this was already deployed on Storj side

**Comment by themylogin** [comment]
@william-gr yes, I signed up for storj with US location and tested this

**Review by yocalebo — APPROVED** [review]
Thanks, lgtm. However, I'm curious if there is a way to get with QE/Storj so we can get some credentials from them for integration tests?

**Review by william-gr — APPROVED** [review]
Terrific, looks good to me (pending caleb's review)

Will this need require any UI change you think?

**Comment by themylogin** [comment]
@william-gr here is UI ticket https://ixsystems.atlassian.net/browse/NAS-117676

@yocalebo I'll create a QE ticket

**Comment by themylogin** [comment]
QE ticket https://ixsystems.atlassian.net/browse/DOCS-4119

**Comment by themylogin** [comment]
backport

**Pull request merged**